modules/src/ics04_channel/packet.rs

```diff
@@ -99,16 +114,84 @@ impl std::fmt::Display for Sequence {
     }
 }
 
+/*
+    A `Packet` represents datagrams flowing between
+    a HOST chain and a COUNTERPARTY chain.
+    It is in fact used in TWO DIRECTIONS: INCOMING and OUTGOING.
+
+    When using a packet, one must identify the HOST chain,
+    the COUNTERPARTY chain, *and* also the DIRECTION.
+    Inside a relaying session, the HOST and COUNTERPARTY
+    chains of the packets remain FIXED and are NEVER FLIPPED.
+
+    Depending on the packet mode, some fields in `Packet`
+    belongs to DIFFERENT chain. This is further noted below.
```

Comment:
This is really confusing to me. Maybe we should discuss this in a call.

Reply:
also adding some comments. We should try to be more in-line with the terms used in the ecosystem (other relayers, specs, docs). For example

* ts-relayer: uses `source | destination` - https://github.com/confio/ts-relayer/blob/main/spec/ibc-setup.md

* golang relayer: uses `source | destination` - https://github.com/cosmos/relayer/blob/112205bab8224295cb046badcda512d37ea1c7b8/cmd/paths.go#L47

and docs also use `chain A | chain B` so I think those are the most popular.

Trying to use terms like `Host | Guest` or `Alpha | Beta` in my opinion might cause some confusion.